src/sentry/workflow_engine/handlers/detector/stateful.py

```diff
@@ -209,9 +214,34 @@ def evaluate_group_key_value(
                     new_substatus=None,
                 )
             else:
-                result, event_data = self.build_occurrence_and_event_data(
-                    group_key, value, PriorityLevel(new_status)
+                detector_occurrence, event_data = self.build_occurrence_and_event_data(
+                    group_key, PriorityLevel(new_status)
                 )
+
+                evidence_data = {
+                    **detector_occurrence.evidence_data,
+                    "detector_id": self.detector.id,
+                    "value": value,
+                }
+
+                result = IssueOccurrence(
+                    str(uuid4()),
+                    self.detector.project_id,
+                    str(uuid4()),
+                    self.build_fingerprint(group_key),
+                    detector_occurrence.issue_title,
+                    detector_occurrence.subtitle,
+                    detector_occurrence.resource_id,
+                    evidence_data,
+                    detector_occurrence.evidence_display or [],
+                    detector_occurrence.type,
+                    datetime.now(),
+                    detector_occurrence.level,
+                    detector_occurrence.culprit or "error",
+                    detector_occurrence.initial_issue_priority or new_status,
+                    detector_occurrence.assignee,
+                )
```

Comment:
Let's write a `from_detector_occurrence` function or something so this is a little cleaner.

We can probable make this cleaner by using kwargs, `IssueOccurrence(**asdict(detector_occurrence), project_id=self.detector.project_Id, ...)`

Reply:
agree about splitting it - this code was meant to simply illustrate pulling data out of the handler :)